pages/home.page.ts

```diff
@@ -0,0 +1,391 @@
+
+        const containsUser = userList.some((user: any) => 
+            user.name === userName && user.phone === userPhone
+        );
+
+        await expect(containsUser).toBe(true);
+    }
+
+    async clickOnEnterBtn() {
+        await this.locators.enterBtn.click();
+    }
+
+    async checkAutorizationFormIsDisplayed() {
+        await expect(this.locators.autorizationForm).toBeVisible();
+    }
+
+    async clickOnSubmitLoginFormBtn() {
+        await this.locators.submitLoginFormBtn.click();
+        await this.page.waitForLoadState('domcontentloaded');
+    }
+
+    async checkInputValue(inputName: string, inputValue: string) {
+        switch(inputName) {
+            case 'email':
+                await expect(await this.locators.loginEmailOrPhoneInput.inputValue()).toBe(inputValue);
+                break
+
+            case 'password':
+                await expect(await this.locators.loginPasswordInput.inputValue()).toBe(inputValue);
+                break
+        }
+    }
+
+    async clickOnHidePasswordIcon() {
+        await this.locators.hidePasswordIcon.click();
+    }
+
+    async checkPasswordInputType(typeName: string) {
+        const passwordInputType = await this.locators.loginPasswordInput.getAttribute('type')
+        switch(typeName){
+            case 'hidden':
+                await expect(passwordInputType).toBe('password');
+                break;
+            
+            case 'shown':
+                await expect(passwordInputType).toBe('text');
+                break;
+        }
+    }
+
+    async checkUserIconIsDisplayed(shouldBeVisible: boolean) {
+        if(shouldBeVisible) {
+            await expect(this.locators.userIcon).toBeVisible();
+        }else {
+            await expect(this.locators.userIcon).not.toBeVisible();
+        }
+    }
+
+    async clickOnUserIcon() {
+        await this.locators.userIcon.click();
+        await this.page.waitForLoadState('domcontentloaded');
+    }
+
+    async checkProfileDropDownIsDisplayed() {
+        await expect(this.locators.profileDropDown).toBeVisible();
+    }
+
+    async checkProfileDropDownEmail(expectedEmail: string) {
+        const currentEmail = await this.locators.profileDropDownEmail.innerText();
+        await expect(currentEmail).toBe(expectedEmail);
+    }
+
+    async logout() {
+        await this.locators.profileLogoutBtn.click();
+    }
+
+    async clickOnMyProfileMenuItem() {
+        await this.locators.myProfileMenuItem.click();
+        await this.page.waitForTimeout(2000);
+    }
+    
+    async checkIncorrectInputFormat() {
+        await this.checkAutorizationFormIsDisplayed();
+        await this.checkUserIconIsDisplayed(false);
+        if (await this.locators.loginErrorInputsMsg.isVisible()) {
+            await expect(this.locators.loginErrorInputsMsg).toBeVisible();
+        } 
+        else if (await this.locators.loginFormErrorMsg.isVisible()) {
+            await expect(this.locators.loginFormErrorMsg).toBeVisible();
```

Comment:
What is this?? This does not make sense. why expect it to be visible if it is already visible???